### PR TITLE
#54 app installer

### LINKED ISSUE #54: Implement an installer with ezplatform.com data
## Description

Project doesn't have any installer to ease project setup
Dump from ezplatform.com are available at this urls : 

- https://ezplatform.com/dumps/ezplatform_page_storage.tar.bz2
- https://ezplatform.com/dumps/ezplatform_page_db.sql.bz2

**Important** : storage file is pretty big as releases files are stored on ezplatform.com
https://github.com/ezsystems/ezplatform-com/issues/66

## Todo

Implement an installer which will download the dump and install it

---
## Initial Issue

After a clean installation with app/console ezplatform:install clean as it's written in INSTALL.md, we have the following error on /.

    Could not find 'eZ\Publish\SPI\Persistence\Content\Type' with identifier 'ID: layout, Status: 0'

@bdunogier gave me this link https://doc.ez.no/pages/viewpage.action?pageId=32867328 but it's not obvious when you arrive on this repo.

I think it can be good to make an installer for that as it's done in the demo project ?

Do you agree on that @SylvainGuittard  ?

+ do you think that we can centralize the install documentation inside this repo instead of pointing to confluence ?

### PR DESCRIPTION
**Pull request by nicolas-bastien**
Fix #54 
Should be merged after #60

## Description

This PR adds an installer which will download data from ezplatform daily backups.

## Test 

Manually from a fresh install

ezplatform-com-installer

### CODE REVIEW

File: src/AppBundle/Installer/AppInstaller.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        $this->output->writeln('');
+        $this->output->writeln(PHP_EOL . '<comment>Import Storage done.</comment>' . PHP_EOL);
+    }
+
+    private function init()
+    {
+        $this->client = new \GuzzleHttp\Client(['base_uri' => 'https://ezplatform.com']);
+
+        $fs = new Filesystem();
+        $fs->mkdir('var/tmp/installer');
```
Comment (clash82) [inline src/AppBundle/Installer/AppInstaller.php]: is it better to use custom tmp instead of the default one?

**Comment by SylvainGuittard** [comment]
thanks @nicolas-bastien for this great contribution.
Just a question: what if the contributeur wants to use windows?

**Comment by clash82** [comment]
>Just a question: what if the contributor wants to use Windows?

This should work also on Windows.

File: src/AppBundle/Installer/AppInstaller.php (hunk 1 above)
Comment (nicolas-bastien) [inline src/AppBundle/Installer/AppInstaller.php]: @clash82 ok you're right, I'm on it

**Comment by nicolas-bastien** [comment]
@clash82 so I refactored this to use OS tmp folder + remove data at the end.

File: .gitignore [hunk 2]
```diff
@@ -33,3 +33,4 @@ composer.lock
 .settings/
 behat.yml
 .php_cs.cache
+/var/tmp
```
Comment (clash82) [inline .gitignore]: you can remove it now

**Comment by clash82** [comment]
@nicolas-bastien maybe we should also delete tmp folder after finishing everything?

File: .gitignore (hunk 2 above)
Comment (nicolas-bastien) [inline .gitignore]: done

File: src/AppBundle/Installer/AppInstaller.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        $this->output->writeln(PHP_EOL . '<info>Moving storage directory contents...</info>');
+        $fs->mkdir('web/var/site/storage');
+        $fs->mirror(
+            $this->tmpFolder . DIRECTORY_SEPARATOR . 'ezplatform_page_storage',
+            'web/var/site/storage'
+        );
+        $progress->advance();
+
+        $this->output->writeln(PHP_EOL . '<info>Cleaning tmp files...</info>');
+        $fs->remove($this->tmpFolder . DIRECTORY_SEPARATOR);
```
Comment (nicolas-bastien) [inline src/AppBundle/Installer/AppInstaller.php]: @clash82 cleaning is done here

Comment (clash82) [inline src/AppBundle/Installer/AppInstaller.php]: You are right, sorry.

File: README.md [hunk 4]
```diff
@@ -6,3 +6,15 @@ eZPlatform.com - eZ Systems' Developer Hub for the Open Source PHP CMS eZ Platfo
 This repo contains the source files for the ezplatform.com website, built on eZ Platform and hosted by the eZ Systems team.
 
 New to eZ Platform? Read more here: https://doc.ez.no/display/DEVELOPER/
+
+## Installation
+
+For more informations about ezplatform installation, please read the [dedicated documentation](https://github.com/ezsystems/ezplatform/blob/master/INSTALL.md).
+ 
+This application provides an installer to set up ezplatform.com data from daily backup.
```
Comment (clash82) [inline README.md]: not `daily` but `weekly` :-)

Comment (nicolas-bastien) [inline README.md]: @clash82 I reword the sentence so if we change the frequency it will always fit

**Comment by nicolas-bastien** [comment]
@clash82 @SylvainGuittard I guess this PR is ready for merge ?

**Comment by clash82** [comment]
ping @SylvainGuittard

**Comment by SylvainGuittard** [comment]
Tested by @clash82 and it works very well. Thanks @nicolas-bastien

**Comment by clash82** [comment]
Thank you @nicolas-bastien, great work!

**Pull request merged**